src/main/java/org/beanpod/switchboard/controller/DeviceController.java

```diff
@@ -53,13 +60,17 @@
 
   @Override
   public ResponseEntity<DeviceModel> createDevice(@Valid CreateDeviceRequest createDeviceRequest) {
-    Optional<DeviceDto> deviceLookup = service.findDevice(createDeviceRequest.getSerialNumber());
+    UserEntity user = userDao.findUser(request.getUserPrincipal().getName());
+    String publicIpAddress = request.getRemoteAddr();
+
+    Optional<DeviceDto> deviceLookup =
```

Comment:
When creating a device with  the same serial number as an existing device that belongs to another user. I was able to "steal" a device from the other user. We probably shouldn't let users choose the ID we used to store devices in the database.

Reply:
Excellent catch, the issue shows that the design of our database does not enforce uniqueness on the serial_number column. Ideally, such transactions would result in a DB exception that we catch and handle in our backend such as "transaction denied". Again, sorry for bringing up area of responsibility. You can see how this PR demands considerable effort because of our decision to tackle restriction later in development. It's kind of like trying to find needles in the haystack.